First pass at preparing for action 1164

aphillips opened the pull request:
Rewrote section 2 and other edits:

- Added link to i18n-glossary
- Fixed my email and attribution
- Created text to address a number of issues from the issues filed against `FindText` back in the day (see commented-out list of issues in the doc)
- Made subsections to address #10 
- Added terminology specific to searching

aphillips committed:
First pass at preparing for #10

Significant editing of the document in preparation for importing some of the material found in the supplied Indic doc. I basically rewrote section 2. This includes starting to bring in the list of issues we filed against `FindText` back in the day.

aphillips committed:
Merge branch 'gh-pages' of https://github.com/aphillips/string-search into gh-pages

xfq commented on index.html:
```diff
@@ -167,16 +167,20 @@ <h3>Goals and Scope</h3>
+        
+        <p class="definition"><dfn>Corpus</dfn> The natural language text contained by a document or set of documents which the user would like to search.</p>
+        <p class="definition"><dfn>Segmentation</dfn> The process of breaking natural language text up into distinct words and phrases. This often includes operations such as "named entity recognition" (such as recognizing that the three word sequence <kbd>Dr. Jonas Salk</kbd> is a person's name).</p>
+        <p class="definition"><dfn>Stemming</dfn> A process or operation that reduces words to their "stem" or root. For example, the words <kbd>runs</kbd>, <kbd>ran</kbd>, and <kbd>running</kbd> all share the stem <kbd>run</kbd>.</p>
```
https://en.wikipedia.org/wiki/Lexeme uses the term lexeme/lemma. I wonder which one is more suitable.

aphillips replied:
`Lemmatization` is also a term used for stemming. I could introduce it as a synonym. I don't want to get too far into the NLP weeds here, though, since these are operations **not** generally found in sub-string matching 😊

xfq commented on index.html:
```diff
@@ -202,21 +206,45 @@ <h2>String Searching in Natural Language Content</h2>
+      
+      <p>One way that sub-string matching usually differs from other types of <a>full-text search</a> is that, while it may use algorithms in an attempt to suppress or ignore textual variations, it does not produce matches that contain additional or unspecified character sequences, words, or phrases.</p>
+      
+      <aside class="example" title="Differences between sub-string matching and other types of search">
```
What are other types of search?

I guess linked data based semantic search and voice search are out of the scope of this document?

aphillips replied:
Yeah, this is sloppy on my part. My intention is to eventually show the difference between a find type operation and classical FTS, e.g. in Find, the term `space monkey` doesn't match `spaced winged monkeys` while in other forms of full-text search it does.

aphillips committed:
Additional edits

Additional edits.

- Removed 'background' section and moved the text about Unicode to the terminology section.
- Removed '...and Notation' from title and contents of that section.
- Added whitespace variation to types of equivalence
- Moved user input section up into 'types of equivalence'
- Added text citing the I18N-Glossary
- Addressed @xfq's comment about the term 'lemmatization'

aphillips committed:
encoding variation examples

Added some examples of encoding variation, including one example from the Bengali section of the doc. These are for discussion.

aphillips committed:
Additional updates and edits

Significant rearrangement of the text and incorporation of more examples.

aphillips committed:
Further edits

aphillips committed:
Update to use document conventions section

Fix to match string-meta. Note minor changes to wording.

aphillips committed:
Reorganize sections and fix descriptions

Significant reordering of sections to make more sense.

Rewrite text and eliminate redundancies.

Fix list of compat/canonical K characters to include character names. Make list complete using UCD v14

aphillips commented:
I'm going to merge this so we can discuss on Thursday--but also because the diff is getting to be "the whole document". If I can get at the Indic doc, I'll start to incorporate those examples in a new PR.
